client/src/pages/ManageSteps/DragDrop.js

```diff
@@ -0,0 +1,220 @@
+import React, { useState, useEffect } from 'react';
+import { DragDropContext, Draggable, Droppable } from 'react-beautiful-dnd';
+import { useHistory } from 'react-router-dom';
+import {
+  Grid as G,
+  Typography as T,
+  Icon,
+  Button,
+  Modal,
+} from '../../components';
+import { buttonStyle } from './style';
+import { navRoutes } from '../../constants';
+
+const columnsFromBackend = {
+  col1: {
+    name: 'Before claiming',
+    stage: 'beforeClaiming',
+    items: [],
+  },
+  col2: {
+    name: 'Claiming',
+    stage: 'claiming',
+    items: [],
+  },
+  col3: {
+    name: 'After claiming',
+    stage: 'afterClaiming',
+    items: [],
+  },
+};
+
+const reorderSteps = (cols) => {
+  let newSteps = [];
+  let cumulativeIndex = 0;
+  Object.keys(cols).forEach((key) => {
+    cols[key].items.forEach((item) => {
+      cumulativeIndex++;
+      newSteps.push({
+        ...item,
+        stage: cols[key].stage,
+        stepOrder: cumulativeIndex,
+      });
+    });
+  });
+  return newSteps;
+};
+
+const onDragEnd = (result, columns, setColumns) => {
+  if (!result.destination) return;
+  const { source, destination } = result;
+
+  if (source.droppableId !== destination.droppableId) {
+    const sourceColumn = columns[source.droppableId];
+    const destColumn = columns[destination.droppableId];
+    const sourceItems = [...sourceColumn.items];
+    const destItems = [...destColumn.items];
+    const [removed] = sourceItems.splice(source.index, 1);
+    destItems.splice(destination.index, 0, removed);
+    setColumns({
+      ...columns,
+      [source.droppableId]: {
+        ...sourceColumn,
+        items: sourceItems,
+      },
+      [destination.droppableId]: {
+        ...destColumn,
+        items: destItems,
+      },
+    });
+  } else {
+    const column = columns[source.droppableId];
+    const copiedItems = [...column.items];
+    const [removed] = copiedItems.splice(source.index, 1);
+    copiedItems.splice(destination.index, 0, removed);
+    setColumns({
+      ...columns,
+      [source.droppableId]: {
+        ...column,
+        items: copiedItems,
+      },
+    });
+  }
+};
+
+function DragDrop({ beforeClaiming, claiming, afterClaiming }) {
+  const [isModalVisible, setIsModalVisible] = useState(false);
+
+  const [columns, setColumns] = useState(columnsFromBackend);
+  const history = useHistory();
+
+  useEffect(() => {
+    setColumns({
+      col1: {
+        name: 'Before claiming',
+        items: beforeClaiming,
+        stage: 'beforeClaiming',
```

Comment:
use the constants data type here so the stage is always consistent with how it's labelled in the database

Reply:
done